cEP-2.md

```diff
@@ -0,0 +1,33 @@
+#Next Gen Bear Design
+| Metadata     | 					                        |
+| ------------ |-----------------------------------------|
+| cEP          | 2                                       |
+| Version      | 2.1                                     |  
+| Title        | Next gen bear design                    |
+| Authors	     | Udayan Tandon <udayan12167@iiitd.ac.in> |
+| Status       | Active                                  |
+| Type         | Process                                 |
+#Abstract
+This document describes how the next generation bears will be designed and implemented.
+
+This cEP describes a new implementation of bears in coala which should make them simpler and make their design more intuitive and better.
+
+#Introduction
+Currently the bear design is clunky and bad. The current design makes global bears dependant on local bears through it's queuing mechanism. It uses communication signals over a queue to indicate that global bears are done with their execution and local bears can begin. Also we have a permanent logger thread that we can do away with.
```

Comment:
Why is the design clunky and bad? be objective, describe exactly what's the issue :+1: